front/pages/api/w/[wId]/assistant/conversations/[cid]/index.ts

```diff
@@ -0,0 +1,104 @@
+import { isLeft } from "fp-ts/lib/Either";
+import * as t from "io-ts";
+import * as reporter from "io-ts-reporters";
+import { NextApiRequest, NextApiResponse } from "next";
+
+import {
+  createConversation,
+  getConversation,
+} from "@app/lib/api/assistant/conversation";
+import { Authenticator, getSession } from "@app/lib/auth";
+import { ReturnedAPIErrorType } from "@app/lib/error";
+import { apiError, withLogging } from "@app/logger/withlogging";
+import { ConversationType } from "@app/types/assistant/conversation";
+
+export type GetConversationsResponseBody = {
+  conversation: ConversationType;
+};
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<
+    GetConversationsResponseBody | ReturnedAPIErrorType | void
+  >
+): Promise<void> {
+  const session = await getSession(req, res);
+  const auth = await Authenticator.fromSession(
+    session,
+    req.query.wId as string
+  );
+
+  const owner = auth.workspace();
+  if (!owner) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace you're trying to modify was not found.",
+      },
+    });
+  }
+
+  if (!auth.user()) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_user_not_found",
+        message: "Could not find the user of the current session.",
+      },
+    });
+  }
+
+  if (!auth.isUser()) {
+    return apiError(req, res, {
+      status_code: 403,
+      api_error: {
+        type: "workspace_auth_error",
+        message:
+          "Only users of the current workspace can access chat sessions.",
+      },
+    });
+  }
+  if (!(typeof req.query.cId === "string")) {
+    return apiError(req, res, {
+      status_code: 400,
+      api_error: {
+        type: "invalid_request_error",
+        message: "Invalid query parameters, `cId` (string) is required.",
+      },
+    });
+  }
+  const conversationId = req.query.cid;
+
+  switch (req.method) {
+    case "GET":
+      const conversation = await getConversation(
+        auth,
+        conversationId as string
```

Comment:
No as you can ensure it’s a string here by setting it to query and testing it’s not null etc